domain/mocks/candidate_route_search_data_worker_mock.go

```diff
@@ -0,0 +1,35 @@
+package mocks
+
+import (
+	"context"
+
+	"github.com/osmosis-labs/sqs/domain"
+)
+
+// CandidateRouteSearchDataWorkerMock is a mock for CandidateRouteSearchDataWorker.
+type CandidateRouteSearchDataWorkerMock struct {
+	ComputeSearchDataSyncFunc  func(ctx context.Context, height uint64, uniqueBlockPoolMetaData domain.BlockPoolMetadata) error
+	ComputeSearchDataAsyncFunc func(ctx context.Context, height uint64, uniqueBlockPoolMetaData domain.BlockPoolMetadata) error
+
+	RegisterListenerFunc func(listener domain.CandidateRouteSearchDataUpdateListener)
```

Comment:
I do not see this being used anywhere. If still unused, why are we adding it?

Reply:
This one is used in the [tests here](https://github.com/osmosis-labs/sqs/blob/d9f566eb739d4a499a6c86ba66e8848997ec9801/ingest/usecase/ingest_usecase_test.go#L437)